kernel_gateway/services/sessions/handlers.py

```diff
@@ -27,6 +27,23 @@ def get(self):
         else:
             super(SessionRootHandler, self).get()
 
+class SessionHandler(TokenAuthorizationMixin,
+                    CORSMixin,
+                    JSONErrorsMixin,
+                    notebook_handlers.SessionHandler):
+    """Extends the notebook session handler with token auth, CORS, and
+    JSON errors.
+    """
+    def set_default_headers(self):
+        self.set_header('Content-Type', 'application/json')
+        self.set_header('Access-Control-Allow-Origin', self.settings['kg_allow_origin'])
+        self.set_header('Access-Control-Allow-Headers', self.settings['kg_allow_headers'])
+        self.set_header('Access-Control-Allow-Methods', self.settings['kg_allow_methods'])
+
+    def options(self, **kwargs):
+        """Method for properly handling CORS pre-flight"""
+        self.finish()
```

Comment:
After working on #384, I'm curious why the [existing `CORSMixin`](https://github.com/jupyter-server/kernel_gateway/blob/7d592e737bf17dddd4464eab9c2f3cd47dd04507/kernel_gateway/mixins.py#L14-L45) isn't already sufficient, since it adds everything that is listed here, except `Content-Type`.

Are the `bases` of the `else` statement below not doing what is intended?